### PR TITLE
Port installation_medium to apypie

### PR DESCRIPTION
**Pull request by mdellweg**
This is a first attempt. At least, it satisfies the existing test playbook without modification.

### CODE REVIEW

File: plugins/modules/foreman_installation_medium.py [hunk 1]
```diff
@@ -133,50 +116,46 @@ def main():
...
+        if not module.desired_absent:  # not 'thin'
+            entities = [module.show_resource('media', entity['id']) for entity in entities]
+    else:
+        entity = module.find_resource_by_name('media', name=entity_dict['name'], failsafe=True)
 
-    if 'locations' in medium_dict:
-        medium_dict['locations'] = find_locations(module, medium_dict['locations'])
+    if not module.desired_absent:
+        if 'operatingsystems' in entity_dict:
+            entity_dict['operatingsystems'] = [module.find_resource('operatingsystems', search='title~"{}"'.format(title), thin=True) for title in entity_dict['operatingsystems']]
```
Comment (evgeni) [inline plugins/modules/foreman_installation_medium.py]:
```
plugins/modules/foreman_installation_medium.py:142:161: E501 line too long (179 > 160 characters)
Makefile:26: recipe for target 'lint' failed
```

File: plugins/modules/foreman_installation_medium.py [hunk 2]
```diff
@@ -133,50 +116,49 @@ def main():
...
 
-    medium_dict = sanitize_entity_dict(medium_dict, name_map)
+        if 'organizations' in entity_dict:
+            entity_dict['organizations'] = module.find_resources('organizations', entity_dict['organizations'], thin=True)
 
     changed = False
     if not affects_multiple:
-        changed = naildown_entity_state(
-            Media, medium_dict, entities, state, module)
+        changed = module.ensure_resource_state('media', entity_dict, entity, module.state, name_map)
```
Comment (evgeni) [inline plugins/modules/foreman_installation_medium.py]: I think `module.state` can now (has to) be removed here

File: plugins/modules/foreman_installation_medium.py [hunk 3]
```diff
@@ -133,50 +116,49 @@ def main():
...
     else:
-        medium_dict.pop('name')
+        entity_dict.pop('name')
         for entity in entities:
-            changed |= naildown_entity_state(
-                Media, medium_dict, entity, state, module)
+            changed |= module.ensure_resource_state('media', entity_dict, entity, module.state, name_map)
```
Comment (evgeni) [inline plugins/modules/foreman_installation_medium.py]: same

**Comment by mdellweg** [comment]
This depends on #332

**Comment by evgeni** [comment]
please rebase. why does the architecture test fail now?

**Pull request merged**